superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -166,7 +167,7 @@ const FilterBar: React.FC<FiltersBarProps> = ({
         };
       });
     },
-    [dataMaskSelected, dispatch, setDataMaskSelected],
+    [JSON.stringify(dataMaskSelected), dispatch, setDataMaskSelected],
```

Comment:
The performance of JSON.stringify can be poor for large objects.
I highly suggest using useRef for dataMaskSelected which can save the cost of JSON stringify as well as the reducing repainting by dataMaskSelected changes.

```suggestion
  const dataMaskSelectedRef = useRef();
  dataMaskSelectedRef.current = dataMaskSelected;
  const handleFilterSelectionChange = useCallback(
    (
      filter: Pick<Filter, 'id'> & Partial<Filter>,
      dataMask: Partial<DataMask>,
    ) => {
      setDataMaskSelected(draft => {
        // force instant updating on initialization for filters with `requiredFirst` is true or instant filters
        if (
          // filterState.value === undefined - means that value not initialized
          dataMask.filterState?.value !== undefined &&
          dataMaskSelectedRef.current[filter.id]?.filterState?.value === undefined &&
          filter.requiredFirst
        ) {
          dispatch(updateDataMask(filter.id, dataMask));
        }

        draft[filter.id] = {
          ...(getInitialDataMask(filter.id) as DataMaskWithId),
          ...dataMask,
        };
      });
    },
    [dispatch, setDataMaskSelected],
  );
```

Reply:
Thanks for response!
Yeah, using ref works as well. I added it to FilterBar and Vertical filter bar instead of JSON.stringifying selected dataMask.